src/pages/my-resources/MyResources.tsx

```diff
@@ -14,11 +15,18 @@ import TabNavigation from '~/components/tab-navigation/TabNavigation'
 import { ResourcesTabsEnum } from '~/types'
 
 const MyResources = () => {
+  const [searchParams, setSearchParams] = useSearchParams()
+  const initialTab = searchParams.get('tab') as ResourcesTabsEnum
   const [activeTab, setActiveTab] = useState<ResourcesTabsEnum>(
-    ResourcesTabsEnum.Lessons
+    initialTab || ResourcesTabsEnum.Lessons
   )
+
   const { t } = useTranslation()
 
+  useEffect(() => {
+    setSearchParams({ tab: activeTab })
+  }, [activeTab, setSearchParams])
```

Comment:
Can we use only `searchParams` to manage state in this component? Because we use two states to handle one functionality and use `useEffect` to sync them.

Reply:
I'd say that this change is necessary because using `useEffect` with state to handle `searchParams` updating often leads to navigation bugs.
Please note the moment when I click the back button several times to go to the previous page.

https://github.com/ita-social-projects/SpaceToStudy-Client/assets/110097862/f85f58e3-2dfa-40c8-af95-71bcdea998b7